packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -551,6 +557,18 @@ class LinearGradient extends Gradient {
 
     return '${objectRuntimeType(this, 'LinearGradient')}(${description.join(', ')})';
   }
+
+  /// Returns a new [LinearGradient] with its colors opacity changed to the given opacity.
+  @override
+  LinearGradient withOpacity(double opacity) {
+    return LinearGradient(
+      begin: begin,
+      end: end,
+      colors: colors.map<Color>((Color color) => color.withOpacity(opacity)).toList(),
+      stops: stops,
```

Comment:
Since we're converting to a list anyway, I think the list compresensions are both faster and more readable:

```suggestion
      colors: <Color>[
        for (Color color in colors)
          color.withOpacity(opacity)
      ],
      stops: stops,
```